File: lib/to_mdx/parsers/open_document/style/drawing_page_properties.rb
# frozen_string_literal: true

require_relative "../base"
require_relative "./fill_image"

module ToMdx
  module Parsers
    module Style
      class DrawingPageProperties < Base
        def initialize(element = nil, context = {})
          super(element, context)
        end

        def to_css
          css_properties = {}

          @element.attributes.each do |name, value|
            case name
            when "draw:fill-color"
              css_properties["background-color"] = value
            when "draw:fill"
              if value == "bitmap"
                css_properties["background-repeat"] = "no-repeat"
                css_properties["background-size"] = "cover"
              end
            when "draw:fill-image-name"
              props = Style::FillImage.fill_image(value)&.to_css
              css_properties.merge!(props) if props
            end
          end

          css_properties
        end

        register_for "style:drawing-page-properties"
      end
    end
  end
end

File: src/parsers/open-document/style/drawing_page_properties.ts
import { BaseElement } from "../../base_element.ts";
import { FillImage } from "./fill_image.ts";

export class DrawingPageProperties extends BaseElement {
  toMdx(): string {
    return "";
  }

  toString(): string {
    return "";
  }

  toCss(): Record<string, string> {
    const cssProperties: Record<string, string> = {};

    Array.from(this.element.attributes).forEach((attr) => {
      const { name, value } = attr;
      switch (name) {
        case "draw:fill-color":
          cssProperties["background-color"] = value;
          break;
        case "draw:fill":
          if (value === "bitmap") {
            cssProperties["background-repeat"] = "no-repeat";
            cssProperties["background-size"] = "cover";
          }
          break;
        case "draw:fill-image-name": {
          const fillImage = FillImage.getFillImage(value);
          if (fillImage) {
            Object.assign(cssProperties, fillImage.toCss());
          }
          break;
        }
      }
    });

    return cssProperties;
  }

  static {
    BaseElement.registerFor.call(this, "style:drawing-page-properties");
  }
}
